08-regression2.Rmd

```diff
@@ -453,41 +463,94 @@ get our test error.
 lm_mult_test_results
 ```
 
-We get that the RMSPE for the multivariate linear regression model 
-of `r format(lm_mult_test_results %>% filter(.metric == 'rmse') %>% pull(.estimate), scientific = FALSE)`. This prediction error
+We obtain an RMSPE for the multivariate linear regression model 
+of `r format(lm_mult_test_results %>% filter(.metric == 'rmse') %>% pull(.estimate), big.mark=",", nsmall=0, scientific = FALSE)`. This prediction error
 is less than the prediction error for the multivariate K-NN regression model,
 indicating that we should likely choose linear regression for predictions of
 house price on this data set. But we should also ask if this more complex
 model is doing a better job of predicting compared to our simple linear
 regression model with only a single predictor (house size). Revisiting last
 section, we see that our RMSPE for our simple linear regression model with
 only a single predictor was 
-`r format(lm_test_results %>% filter(.metric == 'rmse') %>% pull(.estimate), scientific = FALSE)`, 
+`r format(lm_test_results %>% filter(.metric == 'rmse') %>% pull(.estimate), big.mark=",", nsmall=0, scientific = FALSE)`, 
 which is slightly more than that of our more complex model. Our model with two predictors
 provided a slightly better fit on test data than our model with just one. 
-
-But should we always end up choosing a model with more predictors than fewer? 
- The answer is no; you never know what model will be the best until you go through the
-process of comparing their performance on held-out test data. Exploratory 
-data analysis can give you some hints, but until you look
-at the prediction errors to compare the models you don't really know.
-Additionally, here we compare test errors purely for the purposes of teaching.
-In practice, when  you want to compare several regression models with
-differing numbers of predictor variables, you should use
-cross-validation on the training set only; in this case choosing the model is part
-of tuning, so you cannot use the test data. There are several well known and more advanced
-methods to do this that are beyond the scope of this course, and they include
-backward or forward selection, and L1 or L2 regularization (also known as Lasso
-and ridge regression, respectively).
+As mentioned earlier, this is not always the case: sometimes including more
+predictors can negatively impact the prediction performance on unseen 
+test data (here we compare test errors purely for the purposes of teaching).
+
+## Designing new predictors
```

Comment:
I like the addition of this section! I wonder though, if we could find some real data... In ISLR, they use the Auto data set to show this:

<img width="784" alt="Screen Shot 2021-05-30 at 9 50 25 PM" src="https://user-images.githubusercontent.com/6865016/120141389-3c3d4880-c191-11eb-94a8-c766b008ad5f.png">

Reply:
minor disagree, in the interest of avoiding "too many datasets". 

the bigger issue is I couldn't find anything that was
- easy to explain quickly
- had an obvious simple structure that we could make a transformation for
- was not `mtcars` to avoid just copying ISLR

Reply:
punting out to a new issue